sbt-scalajs-bundler/src/main/scala-sjs-1.x/scalajsbundler/sbtplugin/Settings.scala

```diff
@@ -22,97 +22,69 @@ private[sbtplugin] object Settings {
   val jsEnvSetting =
     jsEnvInput := jsEnvInput.dependsOn(npmUpdate).value
 
-  // Override Scala.js setting, which does not support the combination of jsdom and CommonJS module output kind
-  val loadedTestFrameworksSetting =
-    loadedTestFrameworks := Def.task {
-      val (env, input) = {
-        Def.taskDyn[(JSEnv, org.scalajs.jsenv.Input)] {
+  val jsEnvTestSettings =
+    Seq(
+      jsEnv := Def.taskDyn[JSEnv] {
+        if (requireJsDomEnv.value) Def.task {
+          val jsdomDir = installJsdom.value
+          new JSDOMNodeJSEnv(JSDOMNodeJSEnv.Config(jsdomDir))
+        } else Def.task {
+          jsEnv.value
```

Comment:
I'm pretty sure this will turn into a cyclic error. A `taskDyn` unfortunately cannot conditionally depend on its previous value. But perhaps it can work with the following rearrangement:
```scala
jsEnv := {
  val defaultJSEnv = jsEnv.value
  Def.taskDyn[JSEnv] {
    if (requireJsDomEnv.value) Def.task {
      val jsdomDir = installJsdom.value
      new JSDOMNodeJSEnv(JSDOMNodeJSEnv.Config(jsdomDir))
    } else Def.task {
      defaultJSEnv
    }
  }
}
```